servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ClientParentChannelInitializer.java

```diff
@@ -22,25 +22,30 @@
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelInboundHandlerAdapter;
 import io.netty.handler.codec.http2.DefaultHttp2GoAwayFrame;
+import io.netty.handler.codec.http2.DefaultHttp2WindowUpdateFrame;
+import io.netty.handler.codec.http2.Http2ConnectionPrefaceAndSettingsFrameWrittenEvent;
 import io.netty.handler.codec.http2.Http2FrameCodecBuilder;
 import io.netty.handler.codec.http2.Http2MultiplexHandler;
-
-import java.util.function.BiPredicate;
+import io.netty.handler.codec.http2.Http2Settings;
 
 import static io.netty.handler.codec.http2.Http2Error.PROTOCOL_ERROR;
 import static io.servicetalk.http.netty.H2ServerParentChannelInitializer.initFrameLogger;
+import static io.servicetalk.http.netty.H2ServerParentChannelInitializer.toNettySettings;
 
 final class H2ClientParentChannelInitializer implements ChannelInitializer {
-
     private final H2ProtocolConfig config;
+    private final io.netty.handler.codec.http2.Http2Settings nettySettings;
 
     H2ClientParentChannelInitializer(final H2ProtocolConfig config) {
         this.config = config;
+        final io.servicetalk.http.api.Http2Settings h2Settings = config.initialSettings();
+        nettySettings = validateClientSettings(toNettySettings(h2Settings));
```

Comment:
I will validate push when setting the protocol, but I will remove validation of max concurrent streams bcz it shouldn't hurt if push is disabled (and it would now require wrapping the object to override, plus not sure we want to override the value anyways as this can be confusing).